api/src/test/java/org/openmrs/CohortMembershipTest.java

```diff
@@ -0,0 +1,160 @@
+	 * {@link BaseContextSensitiveTest} is run right before this method.
+	 *
+	 * @throws Exception
+	 */
+	@Before
+	public void runBeforeAllTests() throws Exception {
+		service = Context.getCohortService();
+	}
+
+	/**
+	 * @verifies return true if as of date is after start date
+	 * @see CohortMembership#isMembershipActive(java.util.Date)
+	 */
+	@Test
+	public void isMembershipActive_shouldReturnTrueIfAsOfDateIsAfterStartDate() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		newMember.setStartDate(startDate);
+
+		Date dateToTest = dateFormat.parse("2017-01-01 12:00:00");
+		assertTrue(newMember.isMembershipActive(dateToTest));
+		assertTrue(newMember.isMembershipActive(new Date()));
+		assertTrue(newMember.isMembershipActive(startDate));
+	}
+
+	/**
+	 * @verifies return false if start date is after as of date
+	 * @see CohortMembership#isMembershipActive(java.util.Date)
+	 */
+	@Test
+	public void isMembershipActive_shouldReturnFalseIfStartDateIsAfterAsOfDate() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		newMember.setStartDate(startDate);
+
+		Date dateToTest = dateFormat.parse("2016-12-01 00:00:00");
+		assertFalse(newMember.isMembershipActive(dateToTest));
+	}
+
+	/**
+	 * @verifies voided membership should be inactive as of a date
+	 * @see CohortMembership#isMembershipActive(java.util.Date)
+	 */
+	@Test
+	public void isMembershipActive_shouldBeInactiveAsOfDateForVoidedMembership() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date endDate = dateFormat.parse("2017-01-31 00:00:00");
+		newMember.setStartDate(startDate);
+		newMember.setEndDate(endDate);
+		newMember.setVoided(true);
+
+		Date dateToTest = dateFormat.parse("2017-02-01 00:00:00");
+		assertFalse(newMember.isMembershipActive(dateToTest));
+	}
+
+	/**
+	 * @verifies compare two membership objects
+	 * @see CohortMembership#compareTo(org.openmrs.CohortMembership)
+	 */
+	@Test
+	public void compareTo_shouldCompareMemberships() throws Exception {
+		CohortMembership aMember = new CohortMembership(new Patient(4));
+		CohortMembership bMember = new CohortMembership(new Patient(4));
+		assertEquals(aMember.compareTo(bMember), 0);
+		assertEquals(bMember.compareTo(aMember), 0);
+	}
+
+	/**
+	 * @verifies compare two objects for equality
+	 * @see CohortMembership#equals(Object)
+	 */
+	@Test
+	public void equals_shouldCompareTwoObjectsForEquality() throws Exception {
+		executeDataSet(COHORT_XML);
```

Comment:
So this is the test that requires adding test data? Do you really need to add it? Cant you just work with  new instances within this method? That way you don't have to make this class context sensitive?